frontend/src/reducer/notificationPage.js

```diff
@@ -52,16 +55,27 @@ export default function notificationPage (state = defaultNotificationsObject, ac
         list: uniqBy([
           serializeNotification(action.notification),
           ...state.list
-        ], 'id')
+        ], 'id'),
+        notificationNotReadCount: state.notificationNotReadCount + 1
       }
     }
 
-    case `${UPDATE}/${NOTIFICATION}`:
-      return { ...state, list: state.list.map(no => no.id === action.notification.id ? action.notification : no) }
+    case `${READ}/${NOTIFICATION}`: {
+      const notification = state.list.find(notification => notification.id === action.notificationId && !notification.read)
+      if (!notification) return
```

Comment:
you are returning undefined.
This is wrong. In this case, you are expressly setting your state to undefined, wiping all previous state.
If you want to do any modification, always `return state` like it is done in the `default` case